ministryofjustice/correspondence_tool_staff
# Ct 3916 select correspondence type page

## Description
Adds new "SAR IR - Subject access request internal review" to the create case page. Clicking this link takes you to the Link case details page for SAR IRs. You can submit a SAR Case number and it will take you to a "Confirm case link" page that has yet to have functionality added to it. It's just a page to confirm that there is a linked case, and will be added to in a future PR.

Notes for reviewer:
- I was a little stuck when I was getting an error with the that there was no :original_case_number attribute on the SAR::InternalReview case type when submitting the case. I couldn't work out why so I added a virtual attribute to allow the form to submit. I wanted to get things working and though I we could discuss a better way to do this at the PR.
- Back links are not working yet will fix in future PR. 

## Self-review checklist



### Screenshots

**Before**
<img width="992" alt="Screenshot 2021-11-09 at 16 32 45" src="https://user-images.githubusercontent.com/13377553/140970663-b23ef1bd-5841-476e-955b-a9ea91fd4941.png">

**After**
<img width="672" alt="Screenshot 2021-11-09 at 16 11 30" src="https://user-images.githubusercontent.com/13377553/140970714-c59453be-fe39-41bf-803e-084cfd683601.png">

Link page:

<img width="775" alt="Screenshot 2021-11-09 at 16 11 59" src="https://user-images.githubusercontent.com/13377553/140970791-88b668b6-3f66-484f-a9ee-da7d3f37cd68.png">

Errors:



<img width="784" alt="Screenshot 2021-11-09 at 16 12 21" src="https://user-images.githubusercontent.com/13377553/140970826-77cd68de-903e-4b97-9632-418c0d52cd07.png">

<img width="873" alt="Screenshot 2021-11-09 at 16 13 24" src="https://user-images.githubusercontent.com/13377553/140970844-996879f1-6026-4412-91fe-4bb7e9611a84.png">

Basic confirm page - not complete yet:

<img width="1026" alt="Screenshot 2021-11-09 at 16 12 52" src="https://user-images.githubusercontent.com/13377553/140970904-52b632c4-121a-46fd-a61e-73888456177c.png">

### Related JIRA tickets
[CT-3916](https://dsdmoj.atlassian.net/browse/CT-3916) 

### Manual testing instructions
- Sign in as London User
- Click "create_case" should see new case link
- click link
- enter: blank, non-sar case num, and check it errors correctly
- then enter sar case num > continue
- check that details are correct on rudimentary details page.
